### PR TITLE
feat(test): transfer domain for LP tokens

### PR DESCRIPTION
**Pull request by chloezxyy**
#### Which issue(s) does this PR fixes?:

Fixes #

### CODE REVIEW

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts [hunk 1]
```diff
@@ -562,7 +605,75 @@ describe('TransferDomain', () => {
...
+        ownerAddress: address
+      }
+      const data = await client.poolpair.createPoolPair(metadata)
+      expect(typeof data).toStrictEqual('string')
+
+      await container.generate(1)
+      const poolpairsAfter = await client.poolpair.listPoolPairs()
+      expect(Object.keys(poolpairsAfter).length).toStrictEqual(
+        poolpairsLengthBefore + 1
+      )
```
Comment (pierregee) [inline packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]:
```suggestion
   
```
No need to assert tests in the creation of poolpair as it's already tested in other places

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts [hunk 2]
```diff
@@ -562,7 +605,75 @@ describe('TransferDomain', () => {
...
-      await expect(promise).rejects.toThrow('Non-DAT or LP tokens are not supported for transferdomain')
+      await expect(promise).rejects.toThrow(
+        'Non-DAT or LP tokens are not supported for transferdomain'
+      )
+    })
+
+    it('(evm -> dvm) should fail if LP token is transferred', async () => {
+      await createToken(container, 'DBTC')
+      const poolpairsBefore = await client.poolpair.listPoolPairs()
+      const poolpairsLengthBefore = Object.keys(poolpairsBefore).length
```
Comment (pierregee) [inline packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]:
```suggestion
```

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts [hunk 3]
```diff
@@ -562,7 +605,75 @@ describe('TransferDomain', () => {
...
+      const address = await container.call('getnewaddress')
+
+      const metadata = {
+        tokenA: 'DFI',
+        tokenB: 'DBTC',
```
Comment (pierregee) [inline packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]:
```suggestion
        tokenB: 'BTC',
```

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts [hunk 4]
```diff
@@ -562,7 +605,75 @@ describe('TransferDomain', () => {
...
+      const poolpairsAfter = await client.poolpair.listPoolPairs()
+      expect(Object.keys(poolpairsAfter).length).toStrictEqual(
+        poolpairsLengthBefore + 1
+      )
+
+      const promise = client.account.transferDomain([
+        {
+          src: {
+            address: evmAddr,
+            amount: '10@DFI-DBTC',
```
Comment (pierregee) [inline packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]:
```suggestion
            amount: '10@DFI-BTC',
```

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts [hunk 5]
```diff
@@ -562,7 +605,75 @@ describe('TransferDomain', () => {
...
+            domain: TransferDomainType.EVM
+          },
+          dst: {
+            address: dvmAddr,
+            amount: '10@DFI-DBTC',
```
Comment (pierregee) [inline packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]:
```suggestion
            amount: '10@DFI-BTC',
```

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts [hunk 6]
```diff
@@ -562,7 +605,75 @@ describe('TransferDomain', () => {
...
+      )
+    })
+
+    it('(dvm -> evm) should fail if LP token is transferred', async () => {
+      // Using created pool pair from "(evm -> dvm) should fail if LP token is transferred" test
+      const promise = client.account.transferDomain([
+        {
+          src: {
+            address: dvmAddr,
+            amount: '10@DFI-DBTC',
```
Comment (pierregee) [inline packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]:
```suggestion
            amount: '10@DFI-BTC',
```

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts [hunk 7]
```diff
@@ -562,7 +605,75 @@ describe('TransferDomain', () => {
...
+            domain: TransferDomainType.DVM
+          },
+          dst: {
+            address: evmAddr,
+            amount: '10@DFI-DBTC',
```
Comment (pierregee) [inline packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]:
```suggestion
            amount: '10@DFI-BTC',
```

**Review by pierregee — CHANGES_REQUESTED** [review]
Also need to check your prettier why it's adding unnecessary code formatting

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 2 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:614]: removed~

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 3 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:619]: has to be DBTC!

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 1 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts:631]: removed

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 2 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: removed~

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 3 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: updated~

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 4 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: updated~

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 5 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: updated~

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 6 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: updated~

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 7 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: updated~

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 1 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: removed

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 2 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: removed

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 3 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: updated

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 4 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: updated

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 6 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: updated

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts (hunk 7 above)
Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: updated

**Review by pierregee — APPROVED** [review]

File: packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts [hunk 8]
```diff
@@ -565,6 +565,61 @@ describe('TransferDomain', () => {
...
+      ])
+      await expect(promise).rejects.toThrow(RpcApiError)
+      await expect(promise).rejects.toThrow(
+        "RpcApiError: 'Test TransferDomainTx execution failed:\n" +
+        "Non-DAT or LP tokens are not supported for transferdomain', code: -32600, method: transferdomain"
+      )
+    })
+
+    it('(dvm -> evm) should fail if LP token is transferred', async () => {
+      // Using created pool pair from "(evm -> dvm) should fail if LP token is transferred" test
```
Comment (pierregee) [inline packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: As discussed, let's move the creation of poolpair in beforeAll to remove dep on previous `it`

Comment (chloezxyy) [reply packages/jellyfish-api-core/__tests__/category/account/transferDomain.test.ts]: sounds good, updated

**Pull request merged**